src/Yarhl.UnitTests/IO/DataStreamTests.cs

```diff
@@ -965,6 +965,41 @@ public void ReadFormAfeterDisposeThrowException()
             Assert.Throws<ObjectDisposedException>(() => stream.ReadFormat<byte>());
         }
 
+        [Test]
+        public void SliceStream()
+        {
+            DataStream stream = new DataStream();
+            stream.WriteByte(0xBE);
+            stream.WriteByte(0xBA);
+            stream.WriteByte(0xCA);
+            stream.WriteByte(0xFE);
+
+            DataStream testSlice = stream.Slice(2);
```

Comment:
Can you add just one more test for the base stream. Something like Assert.AreSame with testSlice.BaseStream and stream.BaseStream? so we know there isn't copies